internal/metadataproviders/openshift/metadata.go

```diff
@@ -81,11 +91,18 @@ func (o *openshiftProvider) Infrastructure(ctx context.Context) (*Infrastructure
 	if err != nil {
 		return nil, err
 	}
-	res := &InfrastructureAPIResponse{}
-	if err := json.NewDecoder(resp.Body).Decode(res); err != nil {
+	data, err := io.ReadAll(resp.Body)
```

Comment:
Can you help me understand why this is needed?  It looks like it does about the same thing

Reply:
Actually it is the same, but it allows to reuse the api response later in the error message.
```go
		return nil, fmt.Errorf("unable to unmarshal response, err: %w, response: %s",
			err, string(data),
```